src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Configuration/BaseProjectConfigurationDimensionProvider.cs

```diff
@@ -0,0 +1,120 @@
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.VS.Editor;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Configuration
+{
+    /// <summary>
+    /// Base project configuration dimension provider
+    /// </summary>
+    internal abstract class BaseProjectConfigurationDimensionProvider : IProjectConfigurationDimensionsProvider2
+    {
+        protected readonly string _dimensionName;
+        protected readonly string _propertyName;
+        protected readonly IProjectXmlAccessor _projectXmlAccessor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BaseProjectConfigurationDimensionProvider"/> class.
+        /// </summary>
+        /// <param name="projectXmlAccessor">Lock service for the project file.</param>
+        /// <param name="dimensionName">Name of the dimension.</param>
+        /// <param name="propertyName">Name of the project property containing the dimension values.</param>
+        public BaseProjectConfigurationDimensionProvider(IProjectXmlAccessor projectXmlAccessor, string dimensionName, string propertyName)
+        {
+            Requires.NotNull(projectXmlAccessor, nameof(projectXmlAccessor));
+            _projectXmlAccessor = projectXmlAccessor;
+
+            _dimensionName = dimensionName;
+            _propertyName = propertyName;
+        }
+
+        /// <summary>
+        /// Gets the property values for the dimension.
+        /// </summary>
+        /// <param name="unconfiguredProject">Unconfigured project.</param>
+        /// <returns>Collection of values for the dimension.</returns>
+        protected virtual async Task<ImmutableArray<string>> GetOrderedPropertyValuesAsync(UnconfiguredProject unconfiguredProject)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+
+            string propertyValue = await GetPropertyValue(unconfiguredProject).ConfigureAwait(true);
+            if (propertyValue == null || string.IsNullOrEmpty(propertyValue))
+            {
+                return ImmutableArray<string>.Empty;
+            }
+            else
+            {
+                return MsBuildUtilities.GetPropertyValues(propertyValue);
+            }
+        }
+
+        /// <summary>
+        /// Gets the defaults values for project configuration dimensions for the given unconfigured project.
+        /// </summary>
+        /// <param name="unconfiguredProject">Unconfigured project.</param>
+        /// <returns>Collection of key/value pairs for the defaults values for the configuration dimensions of this provider for given project.</returns>
+        public virtual async Task<IEnumerable<KeyValuePair<string, string>>> GetDefaultValuesForDimensionsAsync(UnconfiguredProject unconfiguredProject)
```

Comment:
It's an exsiting CPS interface so there's not much we can change there without making a changeo n the interface. I know this is extra confusing though I had to debug through it a few times to understand what it expects. The reason why it's an IEnumerable instead of a single value is that your provider can return multiple dimensions. So for example if instead of having one provider for config and one for platform we could have one that does both and you'd need to return at that point 2 values, one for the default config and one for the default platform.